ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -144,17 +146,31 @@ private OperationController getOperationController() {
       throw new IllegalArgumentException("blobId or options must not be null");
     }
     currentOperationsCount.incrementAndGet();
+    final FutureResult<GetBlobResult> futureResult = new FutureResult<>();
+    GetBlobOptionsInternal internalOptions = new GetBlobOptionsInternal(options, false, routerMetrics.ageAtGet);
+    boolean isEncrypted = false;
+    try {
+      isEncrypted = new BlobId(blobId, clusterMap).isEncrypted();
+    } catch (IOException e) {
+      RouterException routerException =
+          new RouterException("Exception thrown during construction of BlobId ", e, RouterErrorCode.InvalidBlobId);
+      completeGetBlobOperation(routerException, internalOptions, futureResult, callback);
+    }
+    Meter blobInfoOperationRate =
```

Comment:
it looks like this code has enough control structures to be put into its own function for readability? something like `trackRate(options, isEncrypted)`

Reply:
Yes, will do move it into separate method